### PR TITLE
fix: report autosynth status to GitHub early and don't crash on reporting

### PR DESCRIPTION
**Pull request by chingor13**
Refactors `autosynth.multi` to update GitHub immediately after running `autosynth.synth` on a single library.

* Suppresses and logs errors if the GitHub issue management APIs fail.
* Fixes the case where if a GitHub API call fails, we crash the `autosynth.multi` run.
* Adds unit tests for all of `autosynth.multi`

### CODE REVIEW

File: autosynth/multi.py [hunk 1]
```diff
@@ -16,83 +18,64 @@
...
-        if returncode == 28:
-            skipped = True
-        else:
-            error = True
+    if library.get("deprecated-execution", False):
+        library_args.append("--deprecated-execution")
 
+    result = executor.run(
+        command + library_args + library.get("args", []) + extra_args,
+        stdout=subprocess.PIPE,
```
Comment (SurferJeffAtGoogle) [inline autosynth/multi.py:56]: Not relevant to this code review, but I notice multi.py is basically swallowing synth.py's output.  Instead, we should pipe it to a file that gets stored in placer.

File: autosynth/multi.py [hunk 2]
```diff
@@ -16,83 +18,64 @@
...
+        stderr=subprocess.STDOUT,
+        check=False,
+        encoding="utf-8",
```
Comment (SurferJeffAtGoogle) [inline autosynth/multi.py:59]: Why encoding=utf-8?

File: autosynth/multi.py [hunk 3]
```diff
@@ -16,83 +18,64 @@
...
+        env=env,
+    )
+    error = result.returncode not in (0, 28)
```
Comment (SurferJeffAtGoogle) [inline autosynth/multi.py:62]: Now that synthtool and autosynth live in the same repo, we can put the magic number 28 into a constant and both code bases can import it.

File: autosynth/multi.py (hunk 2 above)
Comment (chingor13) [reply autosynth/multi.py:59]: I'm not entirely sure, but it was already there.

File: autosynth/multi.py [hunk 4]
```diff
@@ -6,93 +6,90 @@
...
-    skipped = False
-    returncode = result["result"].returncode
-
-    if returncode != 0:
-        if returncode == 28:
-            skipped = True
-        else:
-            error = True
-
+def _execute(command: typing.List[str], env: typing.Any) -> typing.Tuple[int, bytes]:
```
Comment (chingor13) [inline autosynth/multi.py:21]: I'm not sure how to reference the [custom type for _ENV](https://github.com/python/typeshed/blob/master/stdlib/3/subprocess.pyi#L37).

**Pull request merged**